kiosk/kiosk_browser/webview_dialog.py

```diff
@@ -0,0 +1,106 @@
+from PyQt5 import QtWidgets, QtCore, QtGui, QtWebEngineWidgets
+
+# Dialog width and height ratio compared to the parent’s size.
+dialog_ratio = 0.8
+
+# Window border thickness
+window_border = 2
+
+def widget(parent, title, url, additional_close_keys, on_dialog_close):
+    """ Embed a web view in a dialog.
+
+        Close with ESC, additional_close_keys, or clicking on the cross.
+    """
+
+    dialog = QtWidgets.QDialog(parent)
+    w = parent.width()
+    h = parent.height()
+    dialog.setGeometry(w * (1 - dialog_ratio) / 2, h * (1 - dialog_ratio) / 2, w * dialog_ratio, h * dialog_ratio)
+
+    overlay = show_overlay(parent)
+    on_close = lambda: close(parent, overlay, dialog, on_dialog_close)
+    show_webview_window(dialog, title, url, on_close)
+
+    QtWidgets.QShortcut('ESC', dialog).activated.connect(on_close)
+    for key in additional_close_keys:
+        QtWidgets.QShortcut(key, dialog).activated.connect(on_close)
+
+    overlay.show()
+    return dialog
+
+def show_overlay(parent):
+    """ Show overlay on all the surface of the parent.
+    """
+
+    widget = QtWidgets.QWidget(parent)
+    widget.setGeometry(0, 0, parent.width(), parent.height())
+    widget.setStyleSheet("background-color: rgba(0, 0, 0, 0.4)")
+    return widget
+
+def show_webview_window(parent, title, url, on_close):
+    """ Show webview window with decorations.
+    """
+
+    widget = QtWidgets.QWidget(parent)
+    widget.setGeometry(0, 0, parent.width(), parent.height())
+    widget.setStyleSheet("background-color: #285577;")
```

Comment:
That comes from Sway, it’s the color of the bar on focus:
https://guix.gnu.org/en/screenshots/sway/

Reply:
I thought it was familiar. :-)

We could use a Dividat shade (#4a5a94 or #183165) or something grayscale to avoid clashes with schemes used in one of the web views.

Reply:
I like those 2 colors, but not in this context. I think the grayscale would be a good solution (using #222222):

![image](https://user-images.githubusercontent.com/6768842/161053343-ac06e8fd-9646-4f6e-b79c-39a58127def6.png)

Reply:
https://github.com/dividat/playos/pull/80/commits/8348a558a11a7f7f8fb5c78e52658fd7c9ddb1d9